src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java

```diff
@@ -93,12 +95,32 @@
     private final RequestConfig.Builder defaultRequestConfigBuilder;
     private final Map<String, String> commonRequestHeaders;
 
+    private SSLContext sslContext;
+
     public IntHttpClient(IntLogger logger, int timeoutInSeconds, boolean alwaysTrustServerCertificate, ProxyInfo proxyInfo) {
-        this(logger, timeoutInSeconds, alwaysTrustServerCertificate, proxyInfo, new BasicCredentialsProvider(), HttpClientBuilder.create(), RequestConfig.custom(), new HashMap<>());
+        this(logger, timeoutInSeconds, alwaysTrustServerCertificate, proxyInfo, SSL_CONTEXT_SUPPLIER.get());
+    }
+
+    public IntHttpClient(IntLogger logger, int timeoutInSeconds, ProxyInfo proxyInfo, SSLContext sslContext) {
+        this(logger, timeoutInSeconds, false, proxyInfo, sslContext);
+    }
+
+    private IntHttpClient(IntLogger logger, int timeoutInSeconds, boolean alwaysTrustServerCertificate, ProxyInfo proxyInfo, SSLContext sslContext) {
```

Comment:
let's move all private constructors next to each other so all public constructors can be together